src/main/scala/edu/ie3/simona/model/participant2/ParticipantModelShell.scala

```diff
@@ -454,13 +430,23 @@ final case class ParticipantModelShell[
   private def determineCurrentState(tick: Long): S = {
     // new state is only calculated if there's an old state and an operating point
     val state = _state
-      .zip(_operatingPoint)
-      .flatMap { case (st, op) =>
-        Option.when(st.tick < tick) {
-          model.determineState(st, op, modelInput)
+      .map { st =>
+        if (st.tick < tick) {
+          // If the state is old, an operating point needs
+          // to be present to determine the curren state
+          model.determineState(
+            st,
+            operatingPoint,
```

Comment:
Is there a reason, you no longer use `.zip(_operatingPoint)`?

Reply:
Yes, zip only gives you a `Some` if both values are present. That was not correct though for the beginning of the simulation

Reply:
Ok.